src/overlays/actors/ovl_En_Daiku2/z_en_daiku2.c

```diff
@@ -22,56 +53,525 @@ const ActorInit En_Daiku2_InitVars = {
+}
+
+void func_80BE6CFC(EnDaiku2* this) {
+    func_80BE6408(this, 3);
+    this->unk_288 = 2;
+    this->actionFunc = func_80BE6D40;
+    this->unk_264 = 0.0f;
+}
+
+void func_80BE6D40(EnDaiku2* this, GlobalContext* globalCtx) {
+    s32 pad[3];
+    s16 sp3A;
+
+    sp3A = Math_Vec3f_Yaw(&this->actor.world.pos, &this->unk_268);
+    if (func_800B84D0(&this->actor, globalCtx) != 0) {
+        this->actionFunc = func_80BE6BC0;
+        return;
+    }
+
+    Math_SmoothStepToS(&this->actor.world.rot.y, sp3A, 1, 0x7D0, 0xA);
+    Math_ApproachF(&this->actor.world.pos.x, this->unk_268.x, 0.5f, fabsf(Math_SinS(this->actor.world.rot.y) * 6.0f));
+    Math_ApproachF(&this->actor.world.pos.z, this->unk_268.z, 0.5f, fabsf(Math_CosS(this->actor.world.rot.y) * 6.0f));
+
+    if ((sqrtf(SQ(this->actor.world.pos.x - this->unk_268.x) + SQ(this->actor.world.pos.z - this->unk_268.z)) < 4.0f) &&
+        (this->unk_258 != 0)) {
+        this->unk_25C += 1;
+        if (this->unk_25C >= this->unk_258->count) {
+            func_80BE6EB0(this);
+            return;
+        }
+        func_80BE61D0(this);
+    }
+    func_800B8614(&this->actor, globalCtx, 80.0f);
+}
+
+void func_80BE6EB0(EnDaiku2* this) {
+    func_80BE6408(this, 2);
+    this->unk_274 = 0;
+    this->unk_288 = 3;
+    this->actionFunc = func_80BE6EF0;
+}
+
+void func_80BE6EF0(EnDaiku2* this, GlobalContext* globalCtx) {
+    f32 sp5C = this->skelAnime.animCurrentFrame;
+    s32 pad[4];
+    Vec3f sp40;
+    s16 var;
+
+    if (func_800B84D0(&this->actor, globalCtx) != 0) {
+        this->actionFunc = func_80BE6BC0;
+        return;
+    }
+
+    Math_SmoothStepToS(&this->actor.world.rot.y, this->actor.home.rot.y, 1, 0xBB8, 0);
+    if (this->unk_284 <= sp5C) {
+        this->unk_274 = 1;
+    }
+
+    if (this->unk_274 != 0) {
+        Math_SmoothStepToS(&this->actor.world.rot.y, Math_Vec3f_Yaw(&this->actor.world.pos, &this->unk_268), 1, 0x7D0,
+                           0xA);
+        Math_ApproachF(&this->actor.world.pos.x, this->unk_268.x, 0.5f,
+                       fabsf(Math_SinS(this->actor.world.rot.y) * 4.0f));
+        Math_ApproachF(&this->actor.world.pos.z, this->unk_268.z, 0.5f,
+                       fabsf(Math_CosS(this->actor.world.rot.y) * 4.0f));
+        if ((sqrtf(SQ(this->actor.world.pos.x - this->unk_268.x) + SQ(this->actor.world.pos.z - this->unk_268.z)) <
+             4.0f) &&
+            (this->unk_258 != 0)) {
+            if (func_80BE64C0(this, globalCtx) == 0) {
+                if (this->unk_276 != 3) {
+                    func_80BE6408(this, 3);
+                }
+
+                if ((this->unk_278 >= 0) && (Flags_GetSwitch(globalCtx, this->unk_278) != 0)) {
+                    this->unk_28A = 5;
+                    if (this->unk_276 != 10) {
+                        func_80BE6408(this, 10);
+                    }
+                    func_80BE65B4(this, globalCtx);
+                    return;
+                }
+
+                this->unk_25C += -1;
```

Comment:
Would something like this match? I know subtraction like this can be weird sometimes.

```suggestion
                this->unk_25C--;
```